hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -691,7 +689,62 @@ public InternalPartition getPartition(int partitionId, boolean triggerOwnerAssig
 
     @Override
     public boolean prepareToSafeShutdown(long timeout, TimeUnit unit) {
-        return partitionReplicaStateChecker.prepareToSafeShutdown(timeout, unit);
+        if (!node.joined()) {
+            return true;
+        }
+
+        if (node.isLiteMember()) {
+            return true;
```

Comment:
We only guarantee partitions' safety (resources related to partitions) during shutdown. Services should handle safety of other non-partition resources by theirselves.